ui/components/app/currency-input/currency-input.js

```diff
@@ -124,17 +124,47 @@ export default function CurrencyInput({
       return;
     }
 
-    const decimalizedHexValue = new Numeric(hexValue, 16)
+    let decimalizedHexValue = new Numeric(hexValue, 16)
       .toBase(10)
       .shiftedBy(assetDecimals)
       .toString();
 
+    const inputValueIfBackspaced = tokenDecimalValue.slice(
```

Comment:
I'm seeing some issues when modifying the input amount:
1. when I try to change 0.5 ETH -> 0.4 ETH, the input becomes 4 ETH
2. inserting a number anywhere besides the end of the amount moves the pointer to the end. this could lead to frustration if the user wants to do something like 2.55 ETH to 1.55 ETH or 1.03425 -> 1.01, for example

video for 1

https://github.com/MetaMask/metamask-extension/assets/100321200/05d1beb9-3f7a-4f74-b6a2-73fac09057ef




video for 2

https://github.com/MetaMask/metamask-extension/assets/100321200/ecb0d86b-d9d7-42b8-bc28-790c3600d703

Reply:
This is buried under a lot of other issues; new PR: #23808